backend/expense/src/main/java/org/egov/digit/expense/repository/querybuilder/BillQueryBuilder.java

```diff
@@ -51,11 +55,50 @@ public String getBillQuery(BillSearchRequest billSearchRequest, List<Object> pre
             addToPreparedStatement(preparedStmtList, ids);
         }
 
-        Set<String> referenceIds = criteria.getReferenceIds();
-        if (!CollectionUtils.isEmpty(referenceIds)) {
+        if (configs.isHealthContextEnabled() && !isValidationSearch) {
+            Set<String> referenceIds = criteria.getReferenceIds();
+            if (!CollectionUtils.isEmpty(referenceIds)) {
+                addClauseIfRequired(query, preparedStmtList);
+                List<String> likeClauses = new ArrayList<>();
+                for (String refId : referenceIds) {
```

Comment:
How many reference IDs can be sent in one search? when doing like query you shouldn't run with an array of values

Reply:
This will only be called for health-context, and as there will be only one referenceId for health, multiple ORs will not be executed, but as referenceIds was a list I have kept it this way instead of doing get(0)